### DISCUSSION TITLE
Removing `magit-commit-diff` from `with-editor-filter-visit-hook` for performance

### DISCUSSION THREAD

**Discussion by joshuaoco** [magit/magit#4773] [Support] [answered]
Hello!

I've noticed in the [Performance](https://magit.vc/manual/magit/Performance.html) section of the manual it suggests removing the commit displaying hook to help out:
```
(remove-hook 'server-switch-hook 'magit-commit-diff)
```

I found that after doing that, the commit window still popped up which appears to be because the function is also registered by `with-editor-filter-visit-hook`` in the magit code.

Removing that hook too does now not display the diff from the commit, and is SUPER fast (I'm TRAMPing to a host ~20ms away), which I'm very happy about.

My question is: Does anyone know if there's some bad consequences to removing this, or has the manual just not been updated with this info?

**Comment by tarsius** [comment, marked as answer]
```
(add-hook 'server-switch-hook 'magit-commit-diff)
(add-hook 'with-editor-filter-visit-hook 'magit-commit-diff)
```

These lines serve the same purpose when different code-paths are taken, so yes, it is safe to undo both. I've updated the manual.
